packages/micropip/src/micropip/_micropip.py

```diff
@@ -81,9 +81,12 @@ async def gather(*coroutines):  # type: ignore[no-redef]
         return result
 
 
-async def _get_pypi_json(pkgname):
+async def _get_pypi_json(pkgname, with_credentials):
     url = f"https://pypi.org/pypi/{pkgname}/json"
-    return json.loads(await fetch_string(url))
+    extra_kwargs = dict()
+    if with_credentials:
+        extra_kwargs["credentials"] = "include"
+    return json.loads(await fetch_string(url, **extra_kwargs))
```

Comment:
Maybe just,
```
async def _get_pypi_json(pkgname, **kwargs):
    url = f"https://pypi.org/pypi/{pkgname}/json"
    return json.loads(await fetch_string(url, **kwargs))
```
we probably don't want to overwrite the value passed by the user, and there could still be a difference between same-origin and ommit?